### PR TITLE
Scan all available networks for public and private IPs

### PR DESCRIPTION
**Pull request by techhat**

### CODE REVIEW

File: salt/cloud/clouds/nova.py [hunk 1]
```diff
@@ -762,14 +762,35 @@ def list_nodes(call=None, **kwargs):
         return {}
     for server in server_list:
         server_tmp = conn.server_show(server_list[server]['id'])[server]
+
+        private = []
+        public = []
+        if not 'addresses' in server_tmp:
```
Comment (hvnsweeting) [inline salt/cloud/clouds/nova.py]: non pythonic -> ``if X not in Y``

File: salt/cloud/clouds/nova.py [hunk 2]
```diff
@@ -762,14 +762,35 @@ def list_nodes(call=None, **kwargs):
...
+            server_tmp = {}
+        for network in server_tmp['addresses'].keys():
```
Comment (hvnsweeting) [inline salt/cloud/clouds/nova.py]: this will not cause KeyError if server_tmp == {} ?

**Comment by techhat** [comment]
Thanks @hvnsweeting, fixed.

I left `.keys()` in because it's explicit. Failure to be explicit is what caused the issues that I was encountering.

**Comment by hvnsweeting** [comment]
> I left .keys() in because it's explicit. Failure to be explicit is what caused the issues that I was encountering.

could you please elaborate about that a bit? as I've never seen any iterate over a dict will fail of that, or because
somehow it passed a list there? then it will raise Exception for attr but I don't see it fit here.
And anw, I saw guys did a lot to fix style and make code py3 compliant, this seems against them

**Pull request merged**